src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3845,6 +4544,14 @@ def spawn(self, launcher, cu):
                 if 'RADICAL_PILOT_PROFILE' in os.environ:
                     launch_script.write("echo pre  stop `%s` >> %s/PROF\n" % (cu['gtod'], cu_tmpdir))
 
+            # YARN pre execution folder permission change
+            # TODO: This needs to move inside the construct command when the launcher
+            #       takes only the CU description
+            if launcher.name == 'LaunchMethodYARN':
+                launch_script.write('\n## Changing Working Directory permissions for YARN\n')
+                launch_script.write('old_perm="`stat -c %a .`"\n')
+                launch_script.write('chmod 777 .\n')
```

Comment:
Here the reason: The YARN container that executes the workload is under the user that Hadoop and Yarn has. It is different from the user that launches the Application. Somehow that user should be able to write to the folder of each CU. That means 2 things. First that the YARN user should have access to the node that the application was launched and 2nd access to the folder of the CU.

There is no guarantee that the user who will run the YARN application will be even in the same group as the YARN user.

Also this 777 changes to the old permission number.

Of course if you have another suggestion I am happy to use it and see how it goes